sympy/tensor/array/array_comprehension.py

```diff
@@ -272,4 +273,59 @@ def _array_subs(arr, var, val):
                     list_gen.append(list_expr.subs(var, val))
                 else:
                     list_gen.append(_array_subs(list_expr, var, val))
-        return ImmutableDenseNDimArray(list_gen)
+        return list_gen
+
+    def tolist(self):
+        """Transform the expanded array to a list
+
+        Raises
+        ======
+
+        ValueError : When there is a symbolic dimension
+
+        Examples
+        ========
+
+        >>> from sympy.tensor.array import ArrayComprehension
+        >>> from sympy import symbols
+        >>> i, j = symbols('i j')
+        >>> a = ArrayComprehension(10*i + j, (i, 1, 4), (j, 1, 3))
+        >>> a.tolist()
+        [[11, 12, 13], [21, 22, 23], [31, 32, 33], [41, 42, 43]]
+        """
+        if self.is_numeric:
```

Comment:
@Upabjojr Regarding this `.is_numeric` property(dicussion raised in #16929), since I am reusing it here, it would be an occasion to modify it.
I have searched over the SymPy API and it is true that this is the only case using `is_numeric`. So maybe it is better to follow the standard. :-)

Reply:
What about "is_shape_numeric"?

Reply:
@Upabjojr That would be a little bit more informative than `is_numeric`. But it is not used in SymPy either. Anyway, I will go with `is_shape_numeric` if you are ok with it. :-)